modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/healthcheck/declaration/PeerDeclarationHealthCheck.scala

```diff
@@ -13,24 +13,28 @@ import scala.concurrent.duration.FiniteDuration
 import scala.reflect.runtime.universe.TypeTag
 
 import org.tessellation.effects.GenUUID
+import org.tessellation.schema.peer.Peer.toP2PContext
 import org.tessellation.schema.peer.PeerId
 import org.tessellation.sdk.config.types.HealthCheckConfig
 import org.tessellation.sdk.domain.cluster.storage.ClusterStorage
 import org.tessellation.sdk.domain.gossip.Gossip
 import org.tessellation.sdk.domain.healthcheck.consensus.HealthCheckConsensus
-import org.tessellation.sdk.domain.healthcheck.consensus.types.ConsensusRounds
+import org.tessellation.sdk.domain.healthcheck.consensus.types.{ConsensusRounds, HealthCheckRoundId}
 import org.tessellation.sdk.infrastructure.consensus._
 import org.tessellation.sdk.infrastructure.consensus.declaration.PeerDeclaration
 
+import io.circe.{Decoder, Encoder}
+
 object PeerDeclarationHealthCheck {
 
-  def make[F[_]: Async: GenUUID, K: TypeTag, A](
+  def make[F[_]: Async: GenUUID, K: TypeTag: Encoder: Decoder, A](
```

Comment:
Maybe a binary codec instead of json? I find it odd that proposals are sent as binary if gossip is used and json if p2p is used.

Reply:
Sounds good, changed :+1: